packages/vscode-extension/src/commands/BaseAction.ts

```diff
@@ -5,56 +5,82 @@
 
 import SuiteCloudRunner from '../core/SuiteCloudRunner';
 import MessageService from '../service/MessageService';
+import VSConsoleLogger from '../loggers/VSConsoleLogger';
 import { VSTranslationService } from '../service/VSTranslationService';
 import { getRootProjectFolder } from '../util/ExtensionUtil';
 import { ERRORS } from '../service/TranslationKeys';
+import { getTimestamp } from '../util/DateUtils';
 
 export default abstract class BaseAction {
-	protected readonly translationService: VSTranslationService;
-	protected executionPath?: string;
-	protected readonly messageService: MessageService;
-	protected readonly commandName: string;
-
-	protected abstract async execute(): Promise<void>;
-
-	constructor(commandName: string) {
-		this.commandName = commandName;
-		this.messageService = new MessageService(this.commandName);
-		this.translationService = new VSTranslationService();
-	}
-
-	protected init() {
-		this.executionPath = getRootProjectFolder();
-	}
-
-	protected validate(): { valid: false; message: string } | { valid: true } {
-		if (!this.executionPath) {
-			return {
-				valid: false,
-				message: this.translationService.getMessage(ERRORS.NO_ACTIVE_FILE_OR_WORKSPACE),
-			};
-		} else {
-			return {
-				valid: true,
-			};
-		}
-	}
-
-	protected async runSuiteCloudCommand(args: { [key: string]: string } = {} ) {
-		return new SuiteCloudRunner(this.executionPath).run({
-			commandName: this.commandName,
-			arguments: args,
-		});
-	}
-
-	public async run() {
-		this.init();
-		const validationStatus = this.validate();
-		if (validationStatus.valid) {
-			return this.execute();
-		} else {
-			this.messageService.showErrorMessage(validationStatus.message);
-			return;
-		}
-	}
+
+    private vsConsoleLogger = new VSConsoleLogger();
+
+    protected readonly translationService: VSTranslationService;
+    protected executionPath?: string;
+    protected readonly messageService: MessageService;
+    protected readonly commandName: string;
+
+    protected abstract async execute(): Promise<void>;
+
+    constructor(commandName: string) {
+        this.commandName = commandName;
+        this.messageService = new MessageService(this.commandName);
+        this.translationService = new VSTranslationService();
+    }
+
+    private logToOutputExecutionDetails(): void {
+        if (this.executionPath) {
+            this.vsConsoleLogger.info(getTimestamp() + " - " + BaseAction.getProjectFolderName(this.executionPath));
+        } else {
+            this.vsConsoleLogger.info(getTimestamp());
+        }
+    }
+
+    private static getProjectFolderName(executionPath: string): string {
+        const executionPathParts = executionPath.replace(/\\/g, "/").split("/");
+
+        return executionPathParts[executionPathParts.length - 1];
+    }
```

Comment:
I don't see the value of making this method static.  Either we place in Util class, which I don't think is really needed, or we use "non static" method.
But maybe I'm missing something here. Let me know your thoughts Karl.

Reply:
Maybe you also want to user `path.sep`.
Check the end of: https://nodejs.org/api/path.html

Reply:
It's a method that doesn't make use of any of the class attributes. So basically it's a utility method that's inside the class.  WebStorm was suggesting to make it static so I thought it should be fine. But since this method is not expected to be used outside of the class, maybe it's a good idea not to put static here. I updated this.

Reply:
Regarding path.sep. It's not really needed here. What I wanted to do is to take the path (regardless if it's in Windows or Linux format) and then just transform all backslash ("\") to slash ("/") so that I don't have to worry about path separator when I split it to get the project folder name.